libs/dataset_deployer.py

```diff
@@ -89,8 +93,11 @@ def write_nested_csv(
     first_row = data[0]
     first_row = {key: value for key, value in first_row.items() if key not in keys_to_skip}
 
-    header = flatten_dict(first_row).keys()
-    header = [column for column in header if column not in keys_to_skip]
+    if columns:
```

Comment:
how about making columns required and removing the branch?

Reply:
I do like the interface of a specified header being non-required, but I did simplify the branch a lot